ui/core/src/model/time.ts

```diff
@@ -104,3 +119,16 @@ export function getSuggestedStepMs(timeRange: AbsoluteTimeRange, width: number)
   const queryRangeMs = timeRange.end.valueOf() - timeRange.start.valueOf();
   return Math.floor(queryRangeMs / width);
 }
+
+/**
+ * Gets the default time range taking into account URL params
+ */
+export function getDefaultTimeRange(from: string, to: string, dashboard?: DashboardResource): TimeRangeValue {
+  const dashboardDuration = dashboard?.spec.duration ?? '1h';
+  const parsedParam = from !== null ? from.split('-')[1] : dashboardDuration;
+  const pastDuration = parsedParam && isDurationString(parsedParam) ? parsedParam : dashboardDuration;
+  if (from === '' || to === '') return { pastDuration };
```

Comment:
That's strange the linter doesn't yell at this syntax.
It's usually a nice practice to put `return ...` in the next line. It will be easier to read